subprojects/docs/src/docs/userguide/feature_variants.adoc

```diff
@@ -0,0 +1,226 @@
+
+- using {metadata-file-spec}[Gradle metadata], everything is published and consumers will get the full benefit of feature variants
+- using POM metadata (Maven), feature variants are published as **optional dependencies** and artifacts of feature variants are published with different _classifiers_
+- using Ivy metadata, feature variants are lost
+====
+
+Publishing feature variants is supported using the `maven-publish` plugin only.
+The Java Plugin (or Java Library Plugin) will take care of registering the additional variants for you, so there's no additional configuration required, only the regular publications:
+
+.Publishing a component with feature variants
+====
+include::sample[dir="java-feature-variant/producer/groovy",files="build.gradle[tags=plugins,publishing]"]
+include::sample[dir="java-feature-variant/producer/kotlin",files="build.gradle.kts[tags=plugins,publishing]"]
+====
+
+[[sec::consuming_feature_variants]]
+== Dependencies on feature variants
+
+[WARNING]
+====
+As mentioned earlier, feature variants can be lossy when published.
+As a consequence, a consumer can depend on a feature variant only in two cases:
+- with a project dependency (in a multi-project build)
+- with Gradle metadata enabled (the publisher MUST have published using Gradle metadata)
+====
+
+A consumer can specify that it needs a specific feature of a producer by declaring required capabilities.
+For example, if a producer declares a "MySQL support" feature like this:
+
+.A library declaring a feature to support MySQL
+====
+include::sample[dir="java-feature-variant/requiring-features/groovy/producer",files="build.gradle[tags=producer]"]
+include::sample[dir="java-feature-variant/requiring-features/kotlin/producer",files="build.gradle.kts[tags=producer]"]
+====
+
+Then the consumer can declare a dependency on the MySQL support feature by doing this:
+
+.Consuming specific features in a multi-project build
+====
+include::sample[dir="java-feature-variant/requiring-features/groovy/consumer",files="build.gradle[tags=consumer]"]
+include::sample[dir="java-feature-variant/requiring-features/kotlin/consumer",files="build.gradle.kts[tags=consumer]"]
+====
+
+Similarly, if the experimental Gradle metadata feature is enabled, it is possible to depend on an external dependency providing feature variants:
+
+.Consuming specific features from an external repository
+====
+include::sample[dir="java-feature-variant/requiring-features-external/groovy",files="build.gradle[tags=consumer]"]
+include::sample[dir="java-feature-variant/requiring-features-external/kotlin",files="build.gradle.kts[tags=consumer]"]
+====
+
+
+[[sec::incompatible_variants]]
+== Handling mutually exclusive variants
+
+The main advantage of using _capabilities_ as a way to handle features is that you can precisely handle compatibility of variants.
+The rule is simple: it's not allowed to have two variants of components that provide the same capability in a dependency graph.
```

Comment:
I would render the part after the ':' using a quote or some other "eye catching" rendering.

Reply:
done